src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Providers/Providers/AzureSqlPsBackupProvider.cs

```diff
@@ -188,7 +188,25 @@ public ProtectionPolicyResponse ModifyPolicy()
 
         public List<Models.AzureRmRecoveryServicesBackupContainerBase> ListProtectionContainers()
         {
-            throw new NotImplementedException();
+            string name = (string)this.ProviderData.ProviderParameters[ContainerParams.Name];
+
+            ProtectionContainerListQueryParams queryParams = new ProtectionContainerListQueryParams();
+
+            queryParams.BackupManagementType = Microsoft.Azure.Management.RecoveryServices.Backup.Models.BackupManagementType.AzureSql.ToString();
+
+            var listResponse = HydraAdapter.ListContainers(queryParams);       
+    
+            List<AzureRmRecoveryServicesBackupContainerBase> containerModels = ConversionHelpers.GetContainerModelList(listResponse);
+
+            if (!string.IsNullOrEmpty(name))
+            {
+                if (containerModels != null)
```

Comment:
![image](https://cloud.githubusercontent.com/assets/1173819/15146210/c3cd5e76-16d7-11e6-83ef-7cac6880b117.png)

As per the implementation of the `GetContainerModelList`, the output list can never be null. So, there's no need for this null check here.